tests/test_qnn.py

```diff
@@ -0,0 +1,257 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for the pennylane.qnn module.
+"""
+import pytest
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras.layers import Layer
+from tensorflow.keras.initializers import RandomNormal
+
+import pennylane as qml
+from pennylane.qnn import KerasLayer
+
+
+@pytest.fixture()
+def get_circuit(n_qubits, output_dim):
+    """Fixture for getting a sample quantum circuit with a controllable qubit number and output
+    dimension. Returns both the circuit and the shape of the weights"""
+
+    dev = qml.device("default.qubit", wires=n_qubits)
+    weight_shapes = {"w1": (3, n_qubits, 3), "w2": (2, n_qubits, 3), "w3": (1,), "w4": 1,
+                     "w5": [3]}
+
+    @qml.qnode(dev, interface='tf')
+    def circuit(w1, w2, w3, w4, w5, x=None):
+        """A circuit that embeds data using the AngleEmbedding and then performs a variety of
+        operations. The output is a PauliZ measurement on the first output_dim qubits."""
+        qml.templates.AngleEmbedding(x, wires=list(range(n_qubits)))
+        if n_qubits > 1:
```

Comment:
TODO: remove the `if` statement when https://github.com/XanaduAI/pennylane/pull/544 is merged in.